spicy/toolchain/include/compiler/detail/codegen/production.h

```diff
@@ -82,7 +87,7 @@ class Meta {
 
 private:
     template<class T>
-    std::optional<T> _value(const NodeRef& n) const {
+    hilti::optional_ref<const T> _value(const NodeRef& n) const {
```

Comment:
Is this needed? The callers of this function usually return `auto` so any external caller likely gets an `optional_ref`. These external callers sometimes store these values, unaware of the lifetime constraints of `optional_ref`.

Reply:
Yeah, the point is to avoid the copy of T, which would mean aren't talking about the same node anymore. Semantics match other uses of `optional_ref`: the callers return references to child nodes, and the assumption is that they stay around for as along as the parent node.